### PR TITLE
Unleash Polling and Delegation V2

### PR DESCRIPTION
**Pull request by louisvdaily**
#### Purpose
- [X] Feature
- [ ] Bug Fix
- [ ] Hotfix
- [ ] Other

#### Details
Added polling and delegation back to client library. Also updated Sample app to remove refresh button added in previous build. Lastly, updated test target to not have two references to files in main Unleash target.

#### Checklist
- [X] Correct base and merge branches

### CODE REVIEW

File: UnleashSampleApp/ViewController.swift [hunk 1]
```diff
@@ -8,9 +8,13 @@
...
 import Unleash
 
-// This constant is to emulate a client knowing its target environment is (e.g. QA, Production)
 struct Constants {
-  static let environment: String = "QA"
+  // This constant is to emulate a client knowing its target environment is (e.g. qa, stage, production)
+  static let environment: String = "qa"
+  static let appName = "ios-unleash-sample-app"
+  static let unleashUrl = "https://unleash.silvercar.com/api"
+  static let featureToggleName = "enable-ios-unleash-sample-app"
```
Comment (deandalm) [inline UnleashSampleApp/ViewController.swift]: Did you create a new toggle in Unleash?

Comment (louisvdaily) [reply UnleashSampleApp/ViewController.swift]: Yes sir I did. One specific for the sample app.

File: unleash/Unleash.swift [hunk 2]
```diff
@@ -77,57 +85,109 @@ public class Unleash {
...
+        case .fulfilled(let response):
+           log("Unleash registered client \(body.instanceId) with response \(response ?? [:])")
+          completion(nil)
+        case .rejected(let error):
+          completion(error)
+        }
+      })
+  }
+  
+  @discardableResult
```
Comment (johnny-1111one1) [inline unleash/Unleash.swift]: I like this notation: Eh, nobody cares.

**Comment by deandalm** [comment]
@louisvdaily mentioned adding missing test coverage but I wasn't sure if he was doing that for this pr or a future one.

**Comment by louisvdaily** [comment]
I’m adding tests cases to this PR. I’ll try to have them added by later tonight.

**Comment by louisvdaily** [comment]
@deandalm @johnny-1111one1 I've added test cases to this PR now.

**Pull request merged**